HousingSearchApi.Tests/V1/E2ETests/Steps/GetAssetSteps.cs

```diff
@@ -204,11 +212,19 @@ public async Task ThenAssetsWithContractApprovalStatusShouldBeIncluded(string co
             result.Results.Assets.Count().Should().Be(expectedNumberOfAssets);
             result.Results.Assets.All(x => x.AssetContract.IsApproved == bool.Parse(contractApprovalStatus));
         }
-        public async Task ThenAllAssetsAreReturned(int amountOfAssets)
+        public async Task ThenAssetsWithProvidedContractStatusShouldBeIncluded(string contractStatus, int expectedNumberOfAssets)
         {
             var resultBody = await _lastResponse.Content.ReadAsStringAsync().ConfigureAwait(false);
             var result = JsonSerializer.Deserialize<APIAllResponse<GetAllAssetListResponse>>(resultBody, _jsonOptions);
-            result.Results.Assets.Count().Should().Be(amountOfAssets);
+
+            result.Results.Assets.Count().Should().Be(expectedNumberOfAssets);
+            result.Results.Assets.All(x => x.AssetContract.IsActive == bool.Parse(contractStatus));
```

Comment:
It's a bit odd having this string to bool parse, can it be prevented, or are the some complications with how this repository handles boolean endpoint inputs?